base/abstractset.jl

```diff
@@ -476,6 +476,67 @@ function isdisjoint(a, b)
     _isdisjoint(a, b)
 end
 
+function isdisjoint(a::AbstractRange{T}, b::AbstractRange{T}) where T
+    (isempty(a) || isempty(b)) && return true
+    fa, la = first(a), last(a)
+    fb, lb = first(b), last(b)
+    if step(a) < zero(step(a))
+        fa, la = la, fa
+    end
+    if step(b) < zero(step(b))
+        fb, lb = lb, fb
+    end
```

Comment:
```suggestion
    fa, _, la = first_step_last_ascending(a)
    fb, _, lb = first_step_last_ascending(b)
```

Reply:
Looks like `first_step_last_ascending` is only defined/valid for `::StepRanges`

Reply:
Right. Even better would be extrema
```suggestion
    fa, la = extrema(a)
    fb, lb = extrema(b)
```